### PR TITLE
WAITP-1209: Map entities part 2

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Re-organise map overlays files
- Add interactive polygon component
- Add bounds to map

---

### Screenshots:
image

image

### CODE REVIEW

File: packages/ui-map-components/src/components/LeafletMap.tsx [hunk 1]
```diff
@@ -217,9 +221,7 @@ export class LeafletMap extends Component<LeafletMapProps> {
 
   flyToBounds = (bounds: LatLngBoundsExpression) => {
     if (!areBoundsValid(bounds)) return;
-    this.map?.fitBounds(bounds, {
-      animate: true,
-    });
+    this.map?.fitBounds(bounds, { animate: true });
```
Comment (acdunham) [inline packages/ui-map-components/src/components/LeafletMap.tsx:224]: Did you end up finding a nice animation for this one?

File: packages/tupaia-web/src/api/queries/useEntities.ts [hunk 2]
```diff
@@ -3,26 +3,31 @@
...
-    } as QueryOptions,
+      enabled,
+    },
   );
 };
+
+export const useEntitiesWithLocation = (projectCode?: string, entityCode?: string) =>
+  useEntities(projectCode, entityCode, {
+    params: { fields: ['parent_code', 'code', 'name', 'type', 'bounds', 'region'] },
+  });
```
Comment (acdunham) [inline packages/tupaia-web/src/api/queries/useEntities.ts:33]: Nice, I like this

**Review by acdunham — APPROVED** [review]
This is looking really nice!

File: packages/ui-map-components/src/components/LeafletMap.tsx (hunk 1 above)
Comment (tcaiger) [reply packages/ui-map-components/src/components/LeafletMap.tsx:224]: I did get it working pretty nicely with map.fitToBounds and it while it was nice and smooth it was much less snappy and a very different feel from how it currently works. So I just went for feature parity and added a note to [the list ](https://beyond-essential.slab.com/posts/web-frontend-rewrite-03l47wk9#h18bd-dev-testing-polishing) of polish as a reminder to look at it later with Edwin.

**Pull request merged**